orttraining/orttraining/core/session/training_session.cc

```diff
@@ -70,9 +82,14 @@ Status SetupOptimizerParams(
         loss_scale_input_name.has_value() ? loss_scale_input_name.value() : "";
     opt_node_config.use_mixed_precision_moments = optimizer_config.use_mixed_precision_moments;
 
-    const auto mixed_precision_weight_name_it = fp32_weight_names_to_mixed_precision_node_args.find(weight_name);
+    const auto mixed_precision_weight_name_it = fp32_weight_names_to_mixed_precision_node_args.find(original_weight_name);
     if (mixed_precision_weight_name_it != fp32_weight_names_to_mixed_precision_node_args.end()) {
       opt_node_config.mixed_precision_weight_arg = mixed_precision_weight_name_it->second;
+    } else {
+      const auto new_mixed_precision_weight_name_it = fp32_weight_names_to_mixed_precision_node_args.find(weight_name);
```

Comment:
if you replace the original_weight_name with weight_name, we will not need this branch

Reply:
changed